fast_response/FastResponseAnalysis.py

```
@@ -992,6 +1020,424 @@ def unblind_TS(self, custom_events=None):
+            mean_flux_arr=fluxToTest
+            nsPerFluxArr=[self.fluxToNs_decDependent(test_flux,d)/test_flux for d in decs_list]
+        else:
+            #work in uniform ns
+            mean_flux_arr=fluxToTest
+            nsPerFluxArr=[1]*12*pow(nside,2) 
+
+
+        for fluxT in mean_flux_arr:
+            temp_val = self.llh.scan(0.0,0.0, scramble = False,spatial_prior=spatial_prior,
+                    time_mask = [self.duration/2., self.centertime],
+                    pixel_scan=[nside, self._pixel_scan_nsigma],
+                    fixed=['nsignal'],
+                    nsignal=np.array([fluxT*conversion for conversion in nsPerFluxArr],dtype=float), 
+
+                    inject=custom_events
+                    )
+            if val is None:
+                val=temp_val
+            else:
+                val=np.hstack((val,temp_val))
+        ras=val["ra"]
+        decs=val["dec"]
+        nss=val["nsignal"]
+    
+        #Placeholder: not required in fixed index analyses, but future improvements could expand this
+        if "gamma" in val.dtype.names:
+            gammas=val["gamma"]
+        
+        TS_spatial_prior_0=val["TS_spatial_prior_0"]
+        idx_max=np.argmax(TS_spatial_prior_0)
+        ramax,decmax=ras[idx_max],decs[idx_max]
+
+        logProbs=TS_spatial_prior_0/2 
+        finalProbs={}
+
+
+        #for each pixel we have lists [ns_1,ns_2,...ns_i], [logP_1,logP_2,...,logP_3]
+        #we want to integrate over ns
+
+        pixel_list_full=list(zip(ras,decs))
+        for pix in set(pixel_list_full):
+            idxes=np.where((ras==pix[0])&(decs==pix[1]))[0]
+            sort_idx=np.argsort(nss[idxes])
+            idxes=idxes[sort_idx]
+            nss_temp=nss[idxes]
+            probs_temp=logProbs[idxes]
+            ns_diff=np.diff(nss_temp)
+            
+            finalProbs[pix]=-np.inf
+            if(len(idxes)!=len(fluxToTest)):
+                print("Not all fluxes have probabilities")
```

can you raise an error here? just make it clearer that the code is exiting and this part failed because of this condition